hca/upload/s3_agent.py

```diff
@@ -53,6 +53,12 @@ def upload_file(self, local_path, target_bucket, target_key, content_type, repor
             self.cumulative_bytes_transferred = 0
             obj.upload_fileobj(fh, **upload_fileobj_args)
 
+    def list_bucket_by_page(self, bucket_name, key_prefix):
+        paginator = self.s3.meta.client.get_paginator('list_objects')
+        for page in paginator.paginate(Bucket=bucket_name, Prefix=key_prefix, PaginationConfig={'PageSize': 100}):
```

Comment:
is there a reason that the default page size is not sufficient?

Reply:
Yes.  Default page size is 1000.  Attempting to get file info on 1000 S3 objects from the Upload Service lambda took a considerable amount of time and was timing out at the API gateway (30s) and sometimes the Lambda timeout too (300s).  It also led to very log pauses in what should be a moderately interactive command.  Reducing to 100 keeps it within the API gateway timeout and makes the command feel more interactive.

I considered getting 1000 at a time from S3 then looping through that 100 at a time to get file info from the Upload Service but frankly I don't think that optimization is worth the extra code complexity.  99% of the work is talking to the Upload Service anyway.

Reply:
Cool, this makes sense, and thanks for the explanation.